src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/ICsDiagnosticWorker.cs

```diff
@@ -1,13 +1,14 @@
 using System.Collections.Immutable;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
+using OmniSharp.Roslyn.CSharp.Services.Diagnostics;
 
 namespace OmniSharp.Roslyn.CSharp.Workers.Diagnostics
 {
     public interface ICsDiagnosticWorker
     {
-        Task<ImmutableArray<(string projectName, Diagnostic diagnostic)>> GetDiagnostics(ImmutableArray<string> documentPaths);
-        Task<ImmutableArray<(string projectName, Diagnostic diagnostic)>> GetAllDiagnosticsAsync();
+        Task<ImmutableArray<DocumentDiagnostics>> GetDiagnostics(ImmutableArray<string> documentPaths);
```

Comment:
This is technically a binary breaking change now that we ship nuget packages. ( https://www.nuget.org/packages/OmniSharp.Roslyn.CSharp/ )

@filipw @savpek 
a) Should we pull this into a different interface?
b) Mark this is a breaking change and v2 OmniSharp?

Reply:
Ah didn't notice that its published with package 🤔 This feature can be implemented without that change however the current `(string projectName, Diagnostic diagnostic)` will face breaking change sooner or later since it's bad fit for public API in extensibility perspective. Another approach may be to add that another interface (..diag..V2 etc) that has new model. 

Personally i vote for breaking change (2.0.x omnisharp). Since at this point there isn't really that many consumers (?) that have to change but adding another supported interface will add maintenance burden and makes public API of omnisharp more complicated 🙂

Reply:
I'm fine with it @filipw ?